waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/mapping/service/PanopticConcurrentOperationExecutorTest.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (C) 2016-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.bdp.waggledance.mapping.service;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.google.common.collect.Lists;
+
+import com.hotels.bdp.waggledance.mapping.model.DatabaseMapping;
+import com.hotels.bdp.waggledance.mapping.service.requests.RequestCallable;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PanopticConcurrentOperationExecutorTest {
+
+  private final static long REQUEST_TIMEOUT = TimeUnit.MILLISECONDS.toMillis(500);
+
+  private @Mock DatabaseMapping mapping1;
+  private @Mock DatabaseMapping mapping2;
+  private @Mock DatabaseMapping mapping3;
+
+  @Before
+  public void setUp() {
+    when(mapping1.getMetastoreMappingName()).thenReturn("mapping1");
+    when(mapping2.getMetastoreMappingName()).thenReturn("mapping2");
+    when(mapping3.getMetastoreMappingName()).thenReturn("mapping3");
+  }
+
+  @Test
+  public void executeRequestsInOrder() throws Exception {
+    PanopticConcurrentOperationExecutor executor = new PanopticConcurrentOperationExecutor();
+    List<DummyRequestCallable> allRequests = Lists
+        .newArrayList(new DummyRequestCallable("call1", mapping1), new DummyRequestCallable("call2", mapping2));
```

Comment:
string sort order could theoretically be why they are sorted below so introduce a call that doesn't sort.